ts/features/pushNotifications/utils/configurePushNotification.ts

```diff
@@ -153,6 +169,32 @@ const messageIdFromPushNotification = (
   return messageIdOnAndroid;
 };
 
+export const deepLinkFromPushNotification = (
+  notification: Omit<ReceivedNotification, "userInfo"> | null
+) => {
+  // Try to decode the notification's payload
+  const payloadDecodeEither = ItwNotificationPayload.decode(notification);
+  if (E.isLeft(payloadDecodeEither)) {
+    // TODO: track the error on Mixpanel
+    return undefined;
+  }
+  const payload = payloadDecodeEither.right;
+
+  // On iOS, the deepLink is stored in the top-level payload
+  const deepLinkOniOS = payload.deepLink;
+  if (deepLinkOniOS != null) {
+    return deepLinkOniOS;
+  }
+  // On Android, the deepLink is stored in the payload.data property
+  const deepLinkOnAndroid = payload.data?.deepLink;
+  if (deepLinkOnAndroid == null) {
+    // TODO: track the error on Mixpanel
+    // In this case, there was no deepLink, so we return
+    return undefined;
+  }
+  return deepLinkOnAndroid;
```

Comment:
We don't have to handle the payload differently for each platform, do we?
In addition to that, I think that if the deeplink is missing we should not consider it an error, for now. But if you prefer we could leave the TODO here (let's add a task for it later, though).

```suggestion
  // Try to decode the notification's payload
  const payloadDecodeEither = ItwNotificationPayload.decode(notification);
  if (E.isLeft(payloadDecodeEither)) {
    return undefined;
  }
  const payload = payloadDecodeEither.right;
  return payload.deepLink || payload.data?.deepLink;
````

What do you think?

Reply:
Agree, fixed here(https://github.com/pagopa/io-app/pull/7521/commits/ef46fa944bf24031dc9b02ca7a82a35b41d5abc7)!